pkg/controller/configurationpolicy/configurationpolicy_controller.go

```diff
@@ -481,32 +481,40 @@ func createInformStatus(mustNotHave bool, numCompliant int, numNonCompliant int,
 	if kind == "" {
 		return
 	}
-	if !mustNotHave && numCompliant == 0 {
-		//noncompliant; musthave and objects do not exist
-		update = createMustHaveStatus(desiredName, kind, nonCompliantObjects, namespaced,
-			plc, indx, compliant)
-	}
-	if mustNotHave && numNonCompliant > 0 {
-		//noncompliant; mustnothave and objects exist
-		update = createMustNotHaveStatus(kind, nonCompliantObjects, namespaced, plc, indx, compliant)
-	}
-	if !mustNotHave && numCompliant > 0 {
-		//compliant; musthave and objects exist
-		compliant = true
-		update = createMustHaveStatus("", kind, compliantObjects, namespaced, plc, indx, compliant)
-	}
-	if mustNotHave && numNonCompliant == 0 {
-		//compliant; mustnothave and no objects exist
-		compliant = true
-		update = createMustNotHaveStatus(kind, compliantObjects, namespaced, plc, indx, compliant)
+
+	if mustNotHave {
+		if numNonCompliant > 0 { // We want no resources, but some were found
+			//noncompliant; mustnothave and objects exist
+			update = createMustNotHaveStatus(kind, nonCompliantObjects, namespaced, plc, indx, compliant)
+		} else if numNonCompliant == 0 {
+			//compliant; mustnothave and no objects exist
+			compliant = true
+			update = createMustNotHaveStatus(kind, compliantObjects, namespaced, plc, indx, compliant)
+		}
+	} else { // !mustNotHave (musthave)
```

Comment:
Is this also the case for `mustonlyhave`? Or is that handled completely separately?